### ISSUE TITLE
Feature Extraction Losses

### ISSUE THREAD

**Issue by tristan-deep** [keras-team/keras#20839]
Current losses supported by Keras are documented [here](https://keras.io/api/losses/). They include probabilistic, regression and hinge losses. Something that would be nice to add are feature extraction losses, such as LPIPS or FID (afaik these are not included in Keras). Alternatively these could be added to the [metrics](https://keras.io/api/metrics/). 

As a starter I have a working Keras implementation of the [LPIPS loss](https://arxiv.org/abs/1801.03924). It relies on:

```python
vgg16 = keras.applications.vgg16.VGG16(include_top=False, weights="imagenet")
````
with some additional layers and it produces the exact same values as the original [lpips package](https://github.com/richzhang/PerceptualSimilarity) (of the authors) as well as the [torch metrics implementation](https://lightning.ai/docs/torchmetrics/stable/image/learned_perceptual_image_patch_similarity.html).

I can make a PR for adding LPIPS if useful, however I need some pointers on how to host the weights (currently I have them on [Hugging Face](https://huggingface.co/tristan-deep/lpips/blob/main/lpips_vgg16.weights.h5).

EDIT: see this #20844 for the PR.

**Comment by Nithurshen** [comment]
Is this still issue open to work? If yes, I would like to work on this.

**Comment by tristan-deep** [comment]
I tried to add it to Keras in #20844, but seems this is more fitting addition to `keras_hub` as explained here:

_We aren't adding any new models to the `keras.applications` API at the moment, since all new models are going to `keras_hub` instead. So I recommend not adding it there._

_Originally posted by @fchollet in https://github.com/keras-team/keras/issues/20844#issuecomment-2661214898_

---

As a sidenote, I did eventually put the LPIPS model in another library that is build on top of Keras 3 ([zea](https://zea.readthedocs.io/)). So this is a [fully Keras 3 compatible LPIPS version](https://zea.readthedocs.io/en/latest/_autosummary/zea.models.lpips.html#module-zea.models.lpips) for people that are looking for that. It could be added to `keras_hub` if there is a clear path and need for that?

```python
# !pip install zea
from zea.models.lpips import LPIPS

model = LPIPS.from_preset("lpips")
```

**Comment by Nithurshen** [comment]
Thank you for the clarification

**Comment by bitloi** [comment]
@hertschuh Can I work this issue?
